site/en/docs/web-platform/compute-pressure/index.md

```diff
@@ -1,264 +1,221 @@
-| 2. Create initial draft of specification | [In Progress](https://github.com/WICG/compute-pressure/)                     |
-| 3. Gather feedback and iterate on design   | In progress                                                                  |
-| 4. Origin trial                          | Complete                                                                     |
-| 5. Launch                                | Not started                                                                  |
+  The Compute Pressure API offers high-level states that represent the pressure on the system.
+  It allows the implementation to use the right underlying hardware metrics to ensure that users
+  can take advantage of all the processing power available to them as long as the system is not
+  under unmanageable stress.
+
+## Current status
+
+<div class="table-wrapper scrollbar">
+
+| Step                                     | Status                              |
+| ---------------------------------------- | ----------------------------------  |
+| 1. Create explainer                      | [Complete][explainer]               |
+| 2. Create initial draft of specification | [Complete][specs]                   |
+| 3. Gather feedback & iterate on design   | [In progress](#feedback)            |
+| 4. Origin trial                          | Not Started                         |
+| 5. Launch                                | Not started                         |
 
 </div>
 
-## API Availability
+### Try out the Compute Pressure API
+
+To experiment with the Compute Pressure API locally, read this [page][how-to].
+
+## Interfaces
+
+Pressure Observer API can be run in the following contexts:
+- Window or main thread
+- Dedicated Worker
+- Shared Worker
+
+The Compute Pressure API defines two new interfaces.
+
+`PressureObserver`: An object to observe the compute pressure of any number of sources at a
+predefined sample rate. Each observer can asynchronously observe pressure changes trends in a
+system.
+
+`PressureRecord`: Describes the pressure trend at a specific moment of transition. Objects of this
+type can only be obtained in two ways: as an input to your PressureObserver callback, or by calling
+PressureObserver.takeRecords().
+
+### PressureObserver
+
+When an `PressureObserver` object is created, it's configured to watch the pressure of supported
+sources, at a given sample rate. The supported sources can be individually observed/unobserved at
+any time during the lifetime of the PressureObserver object. The sample rate cannot be changed
+after the creation of the object.
+
+#### Constructor
+
+`PressureObserver(callback, options)`: Creates a new `PressureObserver` object which will invoke a
+specified callback function when it detects that a change in the values of the source being
+observed has happened.
+
+The constructor takes a [callback](#callback) function and [options](#options)(optional), as parameters.
```

Comment:
The constructor takes a mandatory [callback](#callback) function and optional [options](#options), as parameters.